### PR TITLE
fix(core): catch hydration marker with implicit body tag

### PR DESCRIPTION
**Pull request by hybrist**
When the browser parses a valid html5 response like this:

```html
<!-- ... -->
<title>My page</title>
</head>
<!--nghm-->
<app-root></app-root>
<!-- ... -->
```

The resulting DOM will only start adding nodes to the body when it runs into the first non-header tag. E.g.:

```yml
- head
  - title "My page"
- comment "nghm"
- body
  - app-root
```

This isn't a sign that comments are modified, so it seems worth to handle it gracefully.

Motivation here is to allow some more minification of the HTML response and easier integration with systems that compose response fragments from different sources.

### CODE REVIEW

**Comment by jkrems** [comment]
Adding some tests because there's a few more edge cases I should handle here.

**Review by AndrewKushnir — APPROVED** [review]

File: packages/core/test/hydration/marker_spec.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {verifySsrContentsIntegrity, SSR_CONTENT_INTEGRITY_MARKER} from '../../src/hydration/utils';
+
+fdescribe('verifySsrContentsIntegrity', () => {
```
[the remark this answers is not in the document]
Comment (jkrems) [reply packages/core/test/hydration/marker_spec.ts]: Oops!
